pkg/declared/resources.go

```diff
@@ -68,6 +70,17 @@ func (r *Resources) Update(ctx context.Context, objects []client.Object, commit
 
 	// Record the declared_resources metric, after parsing but before validation.
 	metrics.RecordDeclaredResources(ctx, commit, len(newObjects))
+	if r.previousCommit != commit && r.previousCommit != "" {
+		// Reset the stream value of previous commit to 0 to provide correct input
```

Comment:
> Reset the stream value of previous commit to 0 to provide correct input for the Otel Collector metricstransform processor's MAX aggregation.

So when we added the commit label to the metric it started keeping the latest value for each commit, instead of the latest value for all commits, right? 

What does that have to do with MAX aggregation?

Reply:
My understanding is that the MAX aggregation would be seeing the `latest value for each commit` since the reconciler pod starts, which means a previous max value persists when user removes objects from SoT.